### PR TITLE
[pytorch] | [build, test] | [ec2, ecs, eks, sagemaker] PT 1.4 patch release

### PR DESCRIPTION
**Pull request by YYStreet**
*Issue #, if available:*

## Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [build] | [test] | [build, test] | [ec2, ecs, eks, sagemaker]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

*Description:*

*Tests run:*

*DLC image/dockerfile:*

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/buildspec-1-5.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
+  account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
+  region: &REGION <set-$REGION-in-environment>
+  framework: &FRAMEWORK pytorch
+  version: &VERSION 1.5.0
```
Comment (TusharKanekiDey) [inline pytorch/buildspec-1-5.yml:4]: Our main buildspec file will be for 1.5 only. 
Changes in 1.4 will far and fewer than PT1.5.
Please create a sperate buildspce for PT1.4 not PT1.5

File: pytorch/buildspec.yml [hunk 2]
```diff
@@ -62,28 +47,6 @@
...
-      build: &PYTORCH_CPU_INFERENCE_PY3 false
-      image_size_baseline: 4899
-      device_type: &DEVICE_TYPE cpu
-      python_version: &DOCKER_PYTHON_VERSION py3
-      tag_python_version: &TAG_PYTHON_VERSION py36
-      tag: !join [ *VERSION, "-", *DEVICE_TYPE, "-", *TAG_PYTHON_VERSION, "-", *OS_VERSION ]
-      docker_file: !join [ docker/, *VERSION, /, *DOCKER_PYTHON_VERSION, /Dockerfile., *DEVICE_TYPE ]
-      context:
-        <<: *INFERENCE_CONTEXT
-    BuildGPUPTInferencePy3DockerImage:
```
Comment (TusharKanekiDey) [inline pytorch/buildspec.yml:76]: PT1.4 have Py2 images as well. Please check if we need that for this release as well

File: pytorch/training/docker/1.4.0/py3/example/Dockerfile.gpu [hunk 3]
```diff
@@ -6,6 +7,6 @@ FROM $BASE_IMAGE
 RUN pip install tensorboardX==1.6.0
 
 # Add any script or repo as required
-RUN cd /var && wget https://raw.githubusercontent.com/kubeflow/pytorch-operator/master/examples/mnist/mnist.py
```
Comment (TusharKanekiDey) [inline pytorch/training/docker/1.4.0/py3/example/Dockerfile.gpu:9]: Please check that this change works or not.
I do not think it would work.

File: pytorch/buildspec.yml (hunk 2 above)
Comment (YYStreet) [reply pytorch/buildspec.yml:76]: py2 images do not contain smdebug

File: pytorch/training/docker/1.4.0/py3/example/Dockerfile.gpu (hunk 3 above)
Comment (YYStreet) [reply pytorch/training/docker/1.4.0/py3/example/Dockerfile.gpu:9]: testing. Not sure about the context we did on PT 1.5 about example image

File: pytorch/buildspec-1-5.yml (hunk 1 above)
Comment (YYStreet) [reply pytorch/buildspec-1-5.yml:4]: What I want to do is to use it as a one time release purpose file. Will recover to PT1.5 once PT 1.4 patch release done.
Which method do you prefer? To do modification on root buildspec.yml? It also requires a recover operation

File: pytorch/training/docker/1.4.0/py3/example/Dockerfile.gpu (hunk 3 above)
Comment (TusharKanekiDey) [reply pytorch/training/docker/1.4.0/py3/example/Dockerfile.gpu:9]: Nvm, it will work.

File: pytorch/buildspec-1-5.yml (hunk 1 above)
Comment (nskool) [reply pytorch/buildspec-1-5.yml:4]: Let's retain 1.5 buildspecs.

File: pytorch/training/docker/1.4.0/py3/example/Dockerfile.gpu [hunk 4]
```diff
@@ -6,6 +7,6 @@ FROM $BASE_IMAGE
...
+COPY mnist.py /var/mnist.py
```
Comment (nskool) [inline pytorch/training/docker/1.4.0/py3/example/Dockerfile.gpu:10]: Why this change?

File: test/dlc_tests/container_tests/bin/pytorch_tests/testPyTorchRegression [hunk 5]
```diff
@@ -5,5 +5,6 @@ BIN_DIR=${HOME_DIR}/bin
 LOG_DIR=${HOME_DIR}/logs
 
 git clone https://github.com/pytorch/examples.git ${HOME_DIR}/artifacts/examples
+sed -i "s/param.add_(-0.1 \* param.grad)/param.data.add_(-0.1 \* param.grad)/g" ${HOME_DIR}/artifacts/examples/regression/main.py # temporary patch for isssue https://github.com/pytorch/examples/issues/782
```
Comment (nskool) [inline test/dlc_tests/container_tests/bin/pytorch_tests/testPyTorchRegression:8]: Why this change?

File: pytorch/training/docker/1.4.0/py3/example/Dockerfile.gpu (hunk 4 above)
Comment (YYStreet) [reply pytorch/training/docker/1.4.0/py3/example/Dockerfile.gpu:10]: This is introduced since PT 1.5. PT team change the script and we now maintain it as a separate artifact  https://github.com/aws/deep-learning-containers/blob/master/pytorch/training/docker/build_artifacts/mnist.py

Comment (saimidu) [reply pytorch/training/docker/1.4.0/py3/example/Dockerfile.gpu:10]: Is this necessary now that we are building a PT 1.4 image?

File: test/dlc_tests/container_tests/bin/pytorch_tests/testPyTorchRegression (hunk 5 above)
Comment (YYStreet) [reply test/dlc_tests/container_tests/bin/pytorch_tests/testPyTorchRegression:8]: There is a issue with this test. The PT example repo fixed a similar issue, but seems forgot to fix this file. https://github.com/pytorch/examples/pull/779

I opened a PR for their repo but did not have response yet. This is a temp approach to make the test pass.

File: pytorch/buildspec-1-5.yml (hunk 1 above)
Comment (YYStreet) [reply pytorch/buildspec-1-5.yml:4]: Thanks for confirming.

File: pytorch/training/docker/1.4.0/py3/example/Dockerfile.gpu (hunk 4 above)
Comment (YYStreet) [reply pytorch/training/docker/1.4.0/py3/example/Dockerfile.gpu:10]: Talked with Tushar. We think it should be a general change and should also apply to PT1.4. @TusharKanekiDey
